pkg/discovery/network/ovnkubernetes.go

```diff
@@ -39,40 +40,117 @@ const (
+		clusterNetwork, err = discoverOvnDBClusterNetwork(ctx, client, ovnDBPod)
+	} else {
+		clusterNetwork, err = discoverOvnNodeClusterNetwork(ctx, client)
+	}
+
+	if err != nil || clusterNetwork == nil {
 		return nil, err
 	}
 
-	err = client.Get(ctx, types.NamespacedName{Namespace: ovnDBPod.Namespace, Name: ovnKubeService}, &corev1.Service{})
+	clusterNetwork.NetworkPlugin = cni.OVNKubernetes
+
+	return clusterNetwork, nil
+}
+
+func discoverOvnDBClusterNetwork(ctx context.Context, client controllerClient.Client, ovnDBPod *corev1.Pod) (*ClusterNetwork, error) {
+	err := client.Get(ctx, types.NamespacedName{Namespace: ovnDBPod.Namespace, Name: ovnKubeService}, &corev1.Service{})
 	if err != nil {
 		return nil, fmt.Errorf("error finding %q service in %q namespace", ovnKubeService, ovnDBPod.Namespace)
 	}
 
-	dbConnectionProtocol := "tcp"
-
-	for i := range ovnDBPod.Spec.Containers {
-		for _, envVar := range ovnDBPod.Spec.Containers[i].Env {
-			if envVar.Name == "OVN_SSL_ENABLE" {
-				if !strings.EqualFold(envVar.Value, "NO") {
-					dbConnectionProtocol = "ssl"
-				}
-			}
-		}
-	}
+	dbConnectionProtocol := findProtocol(ovnDBPod)
 
 	clusterNetwork := &ClusterNetwork{
-		NetworkPlugin: cni.OVNKubernetes,
 		PluginSettings: map[string]string{
 			OvnNBDB: fmt.Sprintf("%s:%s.%s:%d", dbConnectionProtocol, ovnKubeService, ovnDBPod.Namespace, OvnNBDBDefaultPort),
 			OvnSBDB: fmt.Sprintf("%s:%s.%s:%d", dbConnectionProtocol, ovnKubeService, ovnDBPod.Namespace, OvnSBDBDefaultPort),
 		},
 	}
 
-	// If the cluster/service CIDRs weren't found we leave it to the generic functions to figure out later
+	updateClusterNetworkFromConfigMap(ctx, client, ovnDBPod.Namespace, clusterNetwork)
+
+	return clusterNetwork, nil
+}
+
+func discoverOvnNodeClusterNetwork(ctx context.Context, client controllerClient.Client) (*ClusterNetwork, error) {
+	// In OVN IC deployments, the ovn DB will be a part of ovnkube-node
+	ovnPod, err := FindPod(ctx, client, "name=ovnkube-node")
+	if err != nil || ovnPod == nil {
+		return nil, err
+	}
+
+	endpointList, err := FindEndpoint(ctx, client, ovnPod.Namespace)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error retrieving the endpoints from namespace %q", ovnPod.Namespace)
+	}
+
+	var clusterNetwork *ClusterNetwork
+
+	if endpointList == nil || len(endpointList.Items) == 0 {
+		clusterNetwork, err = createLocalClusterNetwork(), nil
+	} else {
+		clusterNetwork, err = createClusterNetworkWithEndpoints(endpointList.Items), nil
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	updateClusterNetworkFromConfigMap(ctx, client, ovnPod.Namespace, clusterNetwork)
+
+	return clusterNetwork, nil
+}
+
+func createLocalClusterNetwork() *ClusterNetwork {
+	return &ClusterNetwork{
+		PluginSettings: map[string]string{
+			OvnNBDB: "local",
+			OvnSBDB: "local",
+		},
+	}
+}
+
+func createClusterNetworkWithEndpoints(endPoints []corev1.Endpoints) *ClusterNetwork {
+	pluginSettings := map[string]string{}
+	var OvnNBDBIPs, OVNSBDBIps string
```

Comment:
The names should be consistently cased and made more distinguishable, *e.g.*

```suggestion
	var northboundDBIPs, southboundDBIPs string
```

(this is all about OVN so the prefix isn’t all that useful).

Reply:
Done,